### ISSUE TITLE
[HOLD] Request money - The user's avatar does not appear in the welcome message

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#11073]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to URL https://staging.new.expensify.com/
and login with any user
2. Click on the FAB button and select request money
3. Enter an amount and choose the new user (you must not have a history of communication with him before)

## Expected Result:
The user's avatar should appear in the welcome message immediately

## Actual Result:
The user's avatar does not appear in the welcome message immediately

## Workaround:
Unknown

## Platform:

Where is this issue occurring?

- Web
- iOS
- Android
- Desktop App
- Mobile Web

**Version Number:** 1.2.1.0

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**Email or phone of affected tester (no customers):** any

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/93399543/190754805-1b343953-ee3a-4304-9e07-d19369e7e213.mp4

**Expensify/Expensify Issue URL:** 

**Issue reported by:** Applause - Internal Team

**Slack conversation:** 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by arosiclair** [comment]
I can reproduce this however having a delay before seeing the avatar/personal details seems like expected behavior. 

@Julesssss @vitHoracek you guys worked on the IOU refactor. Do you think this delay in seeing the avatar for a new chat is expected?

**Comment by Julesssss** [comment]
> @Julesssss @vitHoracek you guys worked on the IOU refactor. Do you think this delay in seeing the avatar for a new chat is expected?

I don't think it's expected, and we should try to fix because it doesn't happen for new chats that begin with a message instead of an IOU.

**Comment by mountiny** [comment]
Hmm, it is not expected, but has there been any PR for this merged already? The CreateIOUTransaction which this problem would be related it -> I am just [reviewing it](https://github.com/Expensify/Web-Expensify/pull/34756) 🤔

**Comment by arosiclair** [comment]
Appreciate the insight I'll take a closer look to see why there's this delay

**Comment by arosiclair** [comment]
So the source of this issue is that the report participants and personal details for them are not populated when the new report is created. The chat view doesn't display these avatars in `ReportActionItemCreated` until those are populated. Eventually when the new chat report is opened, we run the deprecated `PersonalDetails_GetForEmails` and `Get?rvl="reportStuff"` commands that sync this data and populate the view. 

Fixing this problem is largely dependent on how the [API refactors for IOUs](https://github.com/Expensify/Expensify/issues/212809) play out. We'll be using Pattern B for this which will optimistically create the new report. To help fix this problem, this optimistic report should have the participants preloaded. A delay for the user's avatar (not the generic one) is probably still expected though as the personal details would need to be synced over Pusher as a response to `RequestMoney` or maybe just included as part of the `OpenReport` command.

@Julesssss @mountiny thoughts on this? It seems like there's no point trying to fix this before the IOU refactors are out. Do you guys think the missing report participants / personal details will be improved with the refactor?

**Comment by mountiny** [comment]
I mean that avatar is about creating a new chat report, which will also be pattern B. We should just make sure the avatar is passed into the optimistic data. We have it since you can see it in the search page. I think we dont have to wait for fixing this on IOU refactors, we might want to see if this refactor will not solve this: https://github.com/Expensify/Expensify/issues/213315

Additionally if this works fine when creating chat (without IOU) I dont see a reason why it should not work when going through the IOU flow 😄

**Comment by mountiny** [comment]
Also fine with that

**Comment by mountiny** [comment]
@MitchExpensify for your interest I think this should be on hold for https://github.com/Expensify/App/pull/11439 and then retest once that PR is done and continue with fix if the issue still prevails

**Comment by MitchExpensify** [comment]
On Vit's rec, putting on hold and will keep an eye on https://github.com/Expensify/App/pull/11439 and retest once that fix is live

**Comment by MitchExpensify** [comment]
Retested and it seems #11439 to have solved this, I can no longer reproduce. Mind checking me on that @kbecciv ? Can you reproduce?

**Comment by mountiny** [comment]
@MitchExpensify Also could not repro, feel free to close this issue.
